### DISCUSSION TITLE
Ideas and proposals for Matefish

### DISCUSSION THREAD

**Discussion by joergoster** [joergoster/Stockfish-old#5] [Ideas]
Feel free to post your ideas and suggestions regarding Matefish here. In English, oder auf deutsch!

**Comment by ginkgo20** [comment]
Hi Jörg!  I am a professional chess player, and I am also interested in chess programming.

I've got some ideas for improving Matefish, but I'm not sure that's possible.

The idea already exists in old versions of Stockfish... but I want to generalization it.

```
    // Internal iterative deepening (~2 Elo)
    if (    depth >= 8 * ONE_PLY
        && !ttMove)
    {
        search<NT>(pos, ss, alpha, beta, depth - 7 * ONE_PLY, cutNode);

        tte = TT.probe(posKey, ttHit);
        ttValue = ttHit ? value_from_tt(tte->value(), ss->ply) : VALUE_NONE;
        ttMove = ttHit ? tte->move() : MOVE_NONE;
    }
```

Will you help me with this?

**Comment by joergoster** [comment]
Well, first of all, thank you for your interest.
Any help/ideas on how to improve Matefish further would be very welcomed.

However, the above code snippet is for getting a best move from the Transposition Table aka Hash Table if none is available when entering a node. That's quite important for move ordering.

Matefish doesn't use a Hash Table right now.
Please also note, that actual development branch is here: https://github.com/joergoster/Stockfish/tree/matefish2

**Comment by ginkgo20** [comment]
Thanks. My goal is to find the way of executing this code.

```
    if ( depth >= 8 * ONE_PLY)
    {
        search<NT>(pos, ss, alpha, beta, depth - 7 * ONE_PLY, cutNode);
    }
```

Can I ask you my questions here?

**Comment by joergoster** [comment]
Not sure I get what you want to do.
If you want to contribute to Stockfish and test your changes, the better start is here: https://github.com/glinscott/fishtest/wiki

Your code won't work in Matefish, which has a much different search than Stockfish.

**Reply by ginkgo20** [reply]
Unfortunately, Admins restricted me in Official_Stockfish just for creating a true pull request and I can't do anything now. 

- https://github.com/official-stockfish/Stockfish/pull/4575

![image](https://github.com/joergoster/Stockfish-old/assets/132969765/752295a7-01b1-40b7-b2d1-eec9e1f54402)

My pull request was completely true because for `PVNode` should be `(beta - alpha > 1)`.

@joergoster Thanks for support and I wish you nothing but success and happiness 💕

**Comment by joergoster** [comment]
> @joergoster Thanks for support and I wish you nothing but success and happiness

@ginkgo20 Thank you and same to you!

**Comment by atumanian** [comment]
@ginkgo20 has told me he was blocked after commenting this thread.
@joergoster did you block him? If so, what is the reason?
He asked me to tell you: "I am sorry and please unblock me".

**Comment by DrGrekenstein** [comment]
Hi @atumanian! I have read your comments on Marco's repository (https://github.com/mcostalba/Stockfish/pull/324).  I know that you have worked for them for many years, but they told you to quit and wouldn't hear you out. I know that "In the darkness you will shine". I know you can if you believe. Good luck!
